network/src/peers/peer_book.rs

```diff
@@ -70,11 +71,11 @@ impl From<SerializedPeerBook> for PeerBook {
 #[derive(Debug, Default)]
 pub struct PeerBook {
     /// The map of the addresses currently being handshaken with.
-    connecting_peers: RwLock<HashSet<SocketAddr>>,
+    connecting_peers: MpmcMap<SocketAddr, Instant>,
```

Comment:
we no longer need an `Instant` to check for any timeouts: the handshake timeout mechanism takes care of this
```suggestion
    connecting_peers: MpmcMap<SocketAddr, ()>,
```